core/src/main/java/org/apache/accumulo/fate/zookeeper/DistributedReadWriteLock.java

```diff
@@ -218,22 +228,69 @@ public boolean tryLock() {
       }
       SortedMap<Long,byte[]> entries = qlock.getEarlierEntries(entry);
       Iterator<Entry<Long,byte[]>> iterator = entries.entrySet().iterator();
-      if (!iterator.hasNext())
+      if (!iterator.hasNext()) {
         throw new IllegalStateException("Did not find our own lock in the queue: " + this.entry
             + " userData " + new String(this.userData, UTF_8) + " lockType " + lockType());
-      return iterator.next().getKey().equals(entry);
+      }
+      if (iterator.next().getKey().equals(entry)) {
+        return true;
+      }
+      if (failBlockers) {
+        // Loop through all of the prior transactions that are waiting to
+        // acquire this write lock. If the transaction has not succeeded or failed,
+        // then fail it and return false from this method so that Utils.reserveX()
+        // will call this method again and will re-check the prior transactions.
+        boolean result = true;
```

Comment:
I was thinking we should drop this and always return false if we are not the first, this seems the safest to avoid concurrency issues.

```suggestion
```

Reply:
So, in the case where we are failingBlockers, WriteLock.tryLock() would always return false and then the FaTE framework would try again. When all of the blockers have been failed and removed, then this txid would be the first in the list. Is that what you are saying? Sounds good to me.

Reply:
Addressed in 2b8fc42ebd19b8c9b3861ab9d5abaa3d34a6888f

Reply:
> Is that what you are saying? Sounds good to me.

yeah that is what I was thinking.  Was thinking if we are not the first in the list that maybe those other fate ops could be running undo operation as part of unwinding themselves in the case of a failure.  Their undo ops could be modifying shared state in ZK and metadata.  So was thinking its safest to wait until we are the first.

Reply:
The changes in 2b8fc42 look good.